### PR TITLE
add links to scope

### PR DESCRIPTION
**Pull request by tatchi**
As discussed, I'll try to implement the structure change in a subsequent PR

### CODE REVIEW

File: src/core/opentelemetry.ml [hunk 1]
```diff
@@ -1007,6 +1016,7 @@ module Trace = struct
     span_id: Span_id.t;
     mutable events: Event.t list;
     mutable attrs: Span.key_value list;
+    mutable links: Span_link.t list;
```
Comment (tatchi) [inline src/core/opentelemetry.ml:1019]: I wonder if we should rename to `Link` to be consistent with `Event` (or the other way around) 🤔

Comment (c-cube) [reply src/core/opentelemetry.ml:1019]: Ah, good point. I think it's fine, `Link.t` would be a bit ambiguous while `Event.t` is fairly clear, imho. I don't particularly want to just break compatibility for that :-). Maybe some time before 1.0 ?

**Review by c-cube — APPROVED** [review]
Thank you, very useful. I think we could also introduce `Scope.make` and use it in otel_cohttp_lwt but it doens't have to be done right now.

File: src/core/opentelemetry.ml [hunk 2]
```diff
@@ -751,6 +751,49 @@ end = struct
...
+    let attributes = List.map _conv_key_value attrs in
+    let dropped_attributes_count =
+      Option.map Int32.of_int dropped_attributes_count
+    in
+    default_span_link
+      ~trace_id:(Trace_id.to_bytes trace_id)
+      ~span_id:(Span_id.to_bytes span_id) ?trace_state ~attributes
+      ?dropped_attributes_count ()
+
+  let[@inline] of_span_ctx ?attrs ctx : t =
```
[the remark this answers is not in the document]
Comment (tatchi) [reply src/core/opentelemetry.ml]: done

**Comment by tatchi** [comment]
> I think we could also introduce `Scope.make` and use it in otel_cohttp_lwt

Added in [`58d1e82` (#66)](https://github.com/imandra-ai/ocaml-opentelemetry/pull/66/commits/58d1e82f667e29a1b7bed5c77416d1ac0a56ccfa)

**Comment by c-cube** [comment]
Thank you, very nice :)

**Pull request merged**